src/main/java/javax/security/jacc/URLPatternSpec.java

```diff
@@ -19,347 +19,334 @@
-	throw new IllegalArgumentException("Invalid URLPatternSpec");
+    private static String getFirstPattern(String urlPatternSpec) {
+        if (urlPatternSpec == null) {
+            throw new IllegalArgumentException("Invalid URLPatternSpec");
+        }
+        
+        int colon = urlPatternSpec.indexOf(":");
+        
+        if (colon < 0) {
+            return urlPatternSpec;
+        }
+        
+        if (colon > 0) {
+            return urlPatternSpec.substring(0, colon);
+        }
+        
+        if (colon == 0) {
+            return EMPTY_STRING;
+        }
+        
+        throw new IllegalArgumentException("Invalid URLPatternSpec");
     }
 
-    private void setURLPatternArray()
-    {
-	if (urlPatternArray == null && urlPatternList != null) {
-
-	    String[] tokens = urlPatternList.split(":",-1);
-
-	    int count = tokens.length;
-
-	    if (count == 0) 
-		throw new IllegalArgumentException
-		    ("colon followed by empty URLPatternList");
-	    urlPatternArray = new URLPattern[count];
-    
-	    int firstType = this.patternType();
-
-	    for (int i=0; i<count; i++) {
-
-		urlPatternArray[i] = new URLPattern(tokens[i]);
-
-		if (urlPatternArray[i].implies(this)) 
-		    throw new IllegalArgumentException
-			("pattern in URLPatternList implies first pattern");
-
-		switch(firstType) {
-		case URLPattern.PT_PREFIX:
-		case URLPattern.PT_EXTENSION:
-		    switch (urlPatternArray[i].patternType()) {
-		    case URLPattern.PT_PREFIX:
-			if (firstType == URLPattern.PT_PREFIX) {
-			    if (super.equals(urlPatternArray[i]) ||
-				!super.implies(urlPatternArray[i]))
-				throw new IllegalArgumentException
-				  ("Invalid prefix pattern in URLPatternList");
-			} 
-			break;
-		    case URLPattern.PT_EXACT:
-			if (!super.implies(urlPatternArray[i]))
-			    throw new IllegalArgumentException
-				("Invalid exact pattern in URLPatternList");
-			break;
-		    default:
-			throw new IllegalArgumentException
-			    ("Invalid pattern type in URLPatternList");
-		    }
-		case URLPattern.PT_DEFAULT:
-		    if (super.equals(urlPatternArray[i]))
-			throw new IllegalArgumentException
-			    ("Invalid default pattern in URLPatternList");
-		    break;
-		case URLPattern.PT_EXACT: 
-		    throw new IllegalArgumentException 
-			("invalid URLPatternSpec");
-		default:
-		    throw new IllegalArgumentException
-			("Invalid pattern type in URLPatternList");
-		}
-	    }
-
-	    Arrays.sort(urlPatternArray);
-
-	    for (int i=0; i<urlPatternArray.length; i++) {
-		if (urlPatternArray[i] != null) {
-		    switch(urlPatternArray[i].patternType()) {
-		    case URLPattern.PT_PREFIX:
-			for (int j=i+1; j<urlPatternArray.length; j++)
-			    if (urlPatternArray[i].implies(urlPatternArray[j]))
-				urlPatternArray[j] = null;
-			break;
-		    default:
-			break;
-		    }
-		}
-	    }
-	}
+    private void setURLPatternArray() {
+        if (urlPatternArray == null && urlPatternList != null) {
```

Comment:
Could `urlPatternArray` and `urlPatternList`be changed to empty arrays instead of nulls without affecting behavior?